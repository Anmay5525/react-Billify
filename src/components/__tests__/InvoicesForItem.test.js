import { mount } from 'enzyme';
import React from 'react';
import ReactDOM, { render } from 'react-dom';
import renderer from 'react-test-renderer';
import InvoicesForItem from '../InvoicesFormItem';

describe('Testing InvoicesForItem component if it ', () => {

    it('renders without crashing', () => {
      const tbody = document.createElement('tbody');
      ReactDOM.render(<InvoicesForItem index={1} data={{}} handleQuantityChange={() => {}} handleDelete={() => {}}></InvoicesForItem>, tbody);
      ReactDOM.unmountComponentAtNode(tbody);
    });
  
    it('matches snapshot', () => {
      const container = mount(<InvoicesForItem index={1} data={{name: 'name', quantity: 0, price: 0, amount: 0}} handleQuantityChange={() => {}} handleDelete={() => {}}/>);
      expect(container.debug()).toMatchSnapshot();
    });
  
  });