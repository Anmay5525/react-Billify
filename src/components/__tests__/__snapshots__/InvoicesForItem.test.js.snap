// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing InvoicesForItem component if it  matches snapshot 1`] = `
<tr
  className="tr"
>
  <td>
    name
  </td>
  <td>
    <input
      className="new-invoice-quantity"
      defaultValue={0}
      max="5"
      min="1"
      name="quantity"
      onChange={[Function]}
      type="number"
    />
  </td>
  <td>
    <span>
      ₹
    </span>
    0
  </td>
  <td>
    <span>
      ₹
    </span>
    0
  </td>
  <td>
    <button
      className="simple-btn"
      onClick={[Function]}
      style={
        Object {
          "fontSize": "0.8em",
        }
      }
      type="button"
    >
      Delete
    </button>
  </td>
</tr>
`;
